### PR TITLE
Fix red screen when damage is disabled

### PR DESCRIPTION
**Pull request by SmallJoker**
- Use a variable for always_fly_fast

Successfully tested with current HEAD on Windows XP, MSVC.

### CODE REVIEW

**Comment by HybridDog** [comment]
@SmallJoker https://github.com/minetest/minetest/pull/2592
This disables damage if it's disabled in your settings, not in server settings, l think https://github.com/minetest/minetest/pull/2592#issuecomment-88944821

File: src/environment.cpp [hunk 1]
```diff
@@ -2116,8 +2117,7 @@ void ClientEnvironment::step(float dtime)
 	*/
 
 	u32 loopcount = 0;
-	do
-	{
+	while (dtime_downcount > 0.001) {
```
Comment (est31) [inline src/environment.cpp]: Replacing `do {} while()` with `while () {}` changes behaviour. Have you taken that into account?

**Comment by est31** [comment]
@HybridDog is right.
I've merged the non-functional part in this commit: b4dfaa3a7a9251cdb0f0867fc29240c808268b1b

**Comment by paramat** [comment]
Zeno and/or hmmmm have a fix for this HP bug introduced by network changes, it's probably best to leave it to them.

**Comment by est31** [comment]
And they are holding it back so that @nerzhul should fix it.

**Comment by SmallJoker** [comment]
Thanks for pushing the working changes; haven't thought about the settings management while trying to fix it.
Let's nerzhul do that for us.

**Comment by kilbith** [comment]
@est31 They're waiting @nerzhul since 3 months (!) now, the fix is not likely to come from him...

**Comment by HybridDog** [comment]
l did a simple and compatible fix https://github.com/minetest/minetest/pull/2592/files?w=1

**Comment by paramat** [comment]
Hmmmm or Zeno will soon fix this, so this should be closed.

**Pull request closed without merging**